nmrfx-processor-gui/src/main/java/org/nmrfx/processor/gui/ProcessorController.java

```diff
@@ -964,30 +1083,38 @@ protected Object call() {
             };
 
             ((Service<Integer>) worker).setOnSucceeded(event -> {
-                processable = true;
-                finishProcessing(processor.getDataset());
-                try {
-                    writeScript(script);
-                } catch (IOException ex) {
-                    GUIUtils.warn("Write Script Error", ex.getMessage());
+                Dataset processedDataset;
+                if (idleMode.get()) {
+                    Dataset dataset = processor.releaseDataset(null);
+                    dataset.script(script);
+                    finishProcessing(dataset);
+                } else {
+                    processedDataset = processor.getDataset();
+                    try {
+                        writeScript(script);
+                    } catch (IOException ex) {
+                        GUIUtils.warn("Write Script Error", ex.getMessage());
+                    }
+                    ConsoleUtil.runOnFxThread(() -> viewDatasetInApp(processedDataset));
                 }
-                setProcessingOff();
-                if (doProcessWhenDone) {
+                processor.getNMRData();
```

Comment:
This seems to just be a getter, and since its not setting a variable, I don't think there is a reason to call it...